### ISSUE TITLE
Actuarial: InsuranceConfig.coinsurance field defined but never applied in claim processing

### ISSUE THREAD

**Issue by AlexFiliakov** [AlexFiliakov/Ergodic-Insurance-Limits#915]
## Description

`InsuranceConfig` at `config/insurance.py:72` defines a `coinsurance` field:

```python
coinsurance: float = Field(default=1.0, gt=0, le=1, description="Coinsurance percentage")
```

However, a search of `insurance_program.py` reveals zero references to `coinsurance`. The `InsuranceProgram.process_claim()` method and `LayerState.apply_claim()` process claims through deductible and layer limits but never apply a coinsurance percentage.

In standard insurance practice, coinsurance means the insured retains a percentage of every covered loss (e.g., 80% coinsurance = insurer pays 80%, insured retains 20% of each covered loss above the deductible). This is distinct from a deductible (which is a fixed dollar threshold).

## Actuarial Impact

Coinsurance is a fundamental coverage modification that affects:
- **Pure premium**: Insurer's expected loss = coinsurance% * layer expected loss
- **Risk transfer**: Lower coinsurance means more risk retained, changing the ergodic benefit calculation
- **Optimal retention**: The optimal coinsurance percentage is a key decision variable alongside deductible and limit
- **Regulatory compliance**: Many commercial property policies have coinsurance clauses (typically 80% or 90%)

Since the HJB optimal control guide (`hjb_solver_guide.rst:369-383`) already envisions coinsurance as a control variable, having it defined but non-functional means optimization results are misleading — the optimizer may vary coinsurance without it having any effect.

## Suggested Fix

Apply coinsurance percentage in `InsuranceProgram.process_claim()` or `LayerState.apply_claim()`:

```python
# After calculating layer payment
layer_payment = layer_payment * self.coinsurance
retained_by_insured += layer_payment * (1 - self.coinsurance)
```

Also ensure the ergodic benefit calculation accounts for the coinsurance retention.

## References

- ISO Commercial Lines Manual, Coinsurance Rules
- Bickelhaupt & Magee, *General Insurance*, Ch. 9 (Coinsurance)
- NAIC Model Laws: Property Insurance Coinsurance Standards

## Acceptance Criteria

- [ ] Coinsurance percentage applied to layer payments in process_claim()
- [ ] ClaimResult includes coinsurance-retained amount
- [ ] Unit test verifies 80% coinsurance pays 80% of layer-covered loss
- [ ] Ergodic benefit correctly reflects retained coinsurance portion

**Comment by AlexFiliakov** [comment]
**Validation (2026-02-13)**: Confirmed. `InsuranceConfig.coinsurance` field defined but never applied in claim processing. Priority medium is appropriate — dead parameter.
